### PR TITLE
PackageTool: add an exclude option

### PR DESCRIPTION
**Pull request by moneromooo**
One may set a number of glob patterns to exclude from the output

### CODE REVIEW

File: Source/Tools/PackageTool/PackageTool.cpp [hunk 1]
```diff
@@ -63,12 +64,33 @@ String ignoreExtensions_[] = {
...
 
+static std::string glob_to_regex(const std::string &val)
+{
+    std::string newval;
+
+    bool escape = false;
+    for (char c: val)
+    {
+      if (c == '*')
+        newval += escape ? "*" : ".*";
```
Comment (1vanK) [inline Source/Tools/PackageTool/PackageTool.cpp:83]: and `escape` stay `true`?

**Comment by 1vanK** [comment]
Have you tested the code?

File: Source/Tools/PackageTool/PackageTool.cpp [hunk 2]
```diff
@@ -63,12 +64,33 @@ String ignoreExtensions_[] = {
...
+
+    bool escape = false;
+    for (char c: val)
+    {
+      if (c == '*')
+        newval += escape ? "*" : ".*", escape = false;
+      else if (c == '?')
+        newval += escape ? "?" : ".", escape = false;
+      else if (c == '\\')
+        newval += '\\', escape = !escape;
```
Comment (1vanK) [inline Source/Tools/PackageTool/PackageTool.cpp:87]: Will `bla\\bla` stay `bla\\bla` ? Or is it by design

**Comment by moneromooo** [comment]
It escapes ? and *, which are the glob characters, and \ so it can be used. There's no canonical way to do that really.

**Comment by 1vanK** [comment]
After your change `\*` will be coverted to regex `.*`. Are you sure exactly what you have in mind?

**Comment by 1vanK** [comment]
`.` wiil be passed to regex as `.`

**Comment by 1vanK** [comment]
In fact, I see no point in checking code that didn't even run.

**Comment by moneromooo** [comment]
Yes, that's a comon character. I don't really want to start escaping everything though. This commit can be dropped if not generic enough.

**Comment by 1vanK** [comment]
Common character `.` in regex not same `.` in path. In path dot is dot.

**Comment by 1vanK** [comment]
>This commit can be dropped if not generic enough.

The problem is that you're wasting people's time reviewing code that you haven't tested yourself.

P.s. code still incorrect. If a character is escaped, you don't really need to check what the character is, but just copy it.

**Comment by moneromooo** [comment]
OK, fair enough.

**Pull request closed without merging**